Sources/Core/X86/HardwareBreakpointManager.cpp

```diff
@@ -216,4 +217,35 @@ ErrorCode HardwareBreakpointManager::isValid(Address const &address,
 
   return super::isValid(address, size, mode);
 }
+
+ErrorCode HardwareBreakpointManager::readDebugRegisters(
```

Comment:
I don't understand the point of this function. Why not simply read the `CPUState` and use that directly instead of using this to copy the data in a vector and use it in this format later on?